### PR TITLE
Added DBT Workflow fixes

### PR DESCRIPTION
**Pull request by OnkarVO7**
### Describe your changes :

Fixed DBT workflow issues
1. Added an empty string check while fetching catalog and run_results file because UI sends an empty string back if the field is kept empty
2. Added condition to skip the analysis and test nodes

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@open-metadata/ingestion

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/database/dbt.py [hunk 1]
```diff
@@ -286,9 +286,6 @@ def yield_data_models(self, dbt_files: DbtFiles) -> Iterable[DataModelLink]:
             self.context.dbt_tests = {}
             for key, manifest_node in manifest_entities.items():
                 try:
-                    # Skip the analysis node since it does not contain relevant metatada
-                    if manifest_node["resource_type"] in ["analysis"]:
-                        continue
 
                     # If the run_results file is passed then only DBT tests will be processed
                     if dbt_files.dbt_run_results:
```
Comment (ayush-shah) [inline ingestion/src/metadata/ingestion/source/database/dbt.py:291]:
```suggestion
                    if dbt_files.dbt_run_results or manifest_node["resource_type"] in ["analysis", "test"]:
```

File: ingestion/src/metadata/utils/dbt_config.py [hunk 2]
```diff
@@ -79,15 +79,20 @@ def _(config: DbtLocalConfig):
             )
             with open(config.dbtManifestFilePath, "r", encoding="utf-8") as manifest:
                 dbt_manifest = manifest.read()
-        if config.dbtRunResultsFilePath is not None:
+        # Need to also check if the file path is an empty string or not
+        # Because UI sends an empty string if nothing is passed in the field
+        if (
+            config.dbtRunResultsFilePath is not None
+            and config.dbtRunResultsFilePath != ""
```
Comment (ayush-shah) [inline ingestion/src/metadata/utils/dbt_config.py:86]:
```suggestion
            if config.dbtRunResultsFilePath:
```

File: ingestion/src/metadata/utils/dbt_config.py [hunk 3]
```diff
@@ -79,15 +79,20 @@ def _(config: DbtLocalConfig):
...
+        ):
             logger.debug(
                 f"Reading [dbtRunResultsFilePath] from: {config.dbtRunResultsFilePath}"
             )
             with open(
                 config.dbtRunResultsFilePath, "r", encoding="utf-8"
             ) as run_results:
                 dbt_run_results = run_results.read()
-        if config.dbtCatalogFilePath is not None:
+        if config.dbtCatalogFilePath is not None and config.dbtCatalogFilePath != "":
```
Comment (ayush-shah) [inline ingestion/src/metadata/utils/dbt_config.py:95]:
```suggestion
        if config.dbtCatalogFilePath:
```

File: ingestion/src/metadata/utils/dbt_config.py [hunk 4]
```diff
@@ -113,7 +118,12 @@ def _(config: DbtHttpConfig):
             config.dbtManifestHttpPath
         )
         dbt_run_results = None
-        if config.dbtRunResultsHttpPath is not None:
+        # Need to also check if the file path is an empty string or not
+        # Because UI sends an empty string if nothing is passed in the field
+        if (
+            config.dbtRunResultsHttpPath is not None
+            and config.dbtRunResultsHttpPath != ""
```
Comment (ayush-shah) [inline ingestion/src/metadata/utils/dbt_config.py:125]:
```suggestion
            config.dbtRunResultsHttpPath:
```

File: ingestion/src/metadata/utils/dbt_config.py [hunk 5]
```diff
@@ -122,7 +132,7 @@ def _(config: DbtHttpConfig):
             )
 
         dbt_catalog = None
-        if config.dbtCatalogHttpPath is not None:
+        if config.dbtCatalogHttpPath is not None and config.dbtCatalogHttpPath != "":
```
Comment (ayush-shah) [inline ingestion/src/metadata/utils/dbt_config.py:135]:
```suggestion
        if config.dbtCatalogHttpPath:
```

**Review by ayush-shah — CHANGES_REQUESTED** [review]

File: ingestion/src/metadata/ingestion/source/database/dbt.py (hunk 1 above)
Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/database/dbt.py:291]: to make it a bit safer we could:

```python
manifest_node("resource_type") in ("analysis", "test")
```
Also, to check `in` requests tuples/sets perform better

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/database/dbt.py:291]: plus lets add some logging if we are skipping logic

**Review by ayush-shah — APPROVED** [review]

**Pull request merged**